pandas/core/indexes/range.py

```diff
@@ -500,13 +500,13 @@ def _minmax(self, meth: str):
 
         return self.start + self.step * no_steps
 
-    def min(self, axis=None, skipna: bool = True, *args, **kwargs) -> int:
+    def min(self, axis=None, skipna: bool = True, *args, **kwargs) -> float:
```

Comment:
This is a bit sad: _minmax can return np.nan which is a float

Reply:
Could we still annotate `int | float` here (and for `_minmax`). I think we had previously discussed wanting to be a bit more explicit with float capturing int

Reply:
Changed it!